docs/network.md

```diff
@@ -0,0 +1,55 @@
+# 10. Message transportation layers.
+
+This section describes different protocols of sending messages currently used in `Near`
+
+## 10.1 Messages between Actors.
+
+`Near` is build on `Actix`'s `actor` framework. (https://actix.rs/book/actix/sec-2-actor.html)
+Usually each actor runs on its own dedicated thread.
+Only messages implementing `actix::Message`, can be sent using between threads.
+Comment on lines +7 to +9
+
+ bowenwang1996 4 days ago Member
+
+We should not leak implementation details into the spec
+
+	Reply…
+Resolve conversation
+
+On example of such message is `PeersRequest` :
+```
+pub struct PeersRequest {}
+impl actix::Message for PeersRequest {
+    type Result = PeerRequestResult;
+}
+pub struct PeerRequestResult {
+    pub peers: Vec<PeerInfo>,
+}
+```
+
+## 10.2 Messages sent through TCP
+Near using `borsh` serialization to exchange messages between nodes (See https://borsh.io/).
+Only messages implementing `BorshSerialize`, `BorshDeserialize` can be sent.
+
+Here is an example of on such message:
+```rust
+#[derive(BorshSerialize, BorshDeserialize)]
```

Comment:
this example is probably not needed (just saying that everything that has a 'derived BorshSerialize/Deserialize' can be sent over the network / written to storage)

Reply:
which means that we have to be VERY careful if we're changing any of those - as it is basically a protocol change.